yb-voyager/cmd/common.go

```diff
@@ -1137,6 +1139,40 @@ type AssessmentReport struct {
 	UnsupportedQueryConstructs []utils.UnsupportedQueryConstruct     `json:"UnsupportedQueryConstructs"`
 }
 
+type UnsupportedFeature struct {
+	FeatureName        string       `json:"FeatureName"`
+	Objects            []ObjectInfo `json:"Objects"`
+	DisplayDDL         bool         `json:"-"` // just used by html format to display the DDL for some feature and object names for other
+	DocsLink           string       `json:"DocsLink,omitempty"`
+	FeatureDescription string       `json:"FeatureDescription,omitempty"`
+}
+
+type ObjectInfo struct {
+	ObjectName   string
+	SqlStatement string
+}
+
+type AssessmentIssue struct {
+	Type         string `json:"Type"`         // Feature, DataType, MigrationCaveat, UQC
+	Name         string `json:"Name"`         // GIN Indexes, Advisory Locks etc
+	Description  string `json:"Description"`  // Based on AssessmentIssue type
+	ObjectName   string `json:"ObjectName"`   // Fully qualified object name(empty if NA, eg UQC)
+	SqlStatement string `json:"SqlStatement"` // DDL or DML(UQC)
+	DocsLink     string `json:"DocsLink"`
+
+	// Store Type-specific details - extensible, can refer any struct
+	Details json.RawMessage `json:"Details,omitempty"`
```

Comment:
How about using `json.Marshaler` type instead of `json.RawMessage` ? that way we can pass a QueryConstuctDetails struct directly to this Details object as long as it implements the `MarshalJSON` method. https://pkg.go.dev/encoding/json#Marshaler

Using RawMessage would mean we have to explicitly marshall the QueryConstuctDetails struct before we create this `AssessmentIssue` struct. 
It's better for the marshal logic to be defined as part of the struct method.